### PR TITLE
Adding error checking to properly catch and handle edge cases

### PR DESCRIPTION
**Pull request by cspensky**
Just a few error conditions that came up in my testing that I cleaned up.

### CODE REVIEW

File: bleak/backends/bluezdbus/client.py [hunk 1]
```diff
@@ -513,6 +513,9 @@ async def read_gatt_char(
             (bytearray) The read data.
 
         """
+        if self._bus is None:
+            raise BleakError("D-Bus object is None")
```
Comment (dlech) [inline bleak/backends/bluezdbus/client.py:517]: How about saying that the client is not connected instead? Isn't that what triggers this problem?

**Review by dlech — COMMENTED** [review]
Thanks for this. You should also try out the `dbus-next-2` branch as it will probably replace the BlueZ backend soon.

I already fixed the last two issues in that branch.

File: bleak/backends/bluezdbus/scanner.py [hunk 2]
```diff
@@ -242,7 +242,10 @@ def _parse_msg(self, message: Message):
             # discovered devices list
 
             msg_path = message.body[0]
-            del self._devices[msg_path]
+            try:
```
Comment (dlech) [inline bleak/backends/bluezdbus/scanner.py:245]: FWIW, here is how I fixed it: https://github.com/hbldh/bleak/pull/389/commits/5bf7266825c7b4714b90e59200739bfc8d014018

Comment (dlech) [reply bleak/backends/bluezdbus/scanner.py:245]: But nothing wrong with this way either :smile:

File: bleak/backends/bluezdbus/client.py (hunk 1 above)
Comment (cspensky) [reply bleak/backends/bluezdbus/client.py:517]: Good point.  Let me look into this one more and clean up the error message.

File: bleak/backends/bluezdbus/scanner.py (hunk 2 above)
Comment (cspensky) [reply bleak/backends/bluezdbus/scanner.py:245]: 👍 I like it.  These were a bit old, I just wanted to document them since they were all short and sweet.

**Comment by cspensky** [comment]
Closing this.  I'll submit another pull request with some cleaned up error messages now that our branches are synced back up.

**Pull request closed without merging**